### ISSUE TITLE
Can you add documentation?

### ISSUE THREAD

**Issue by ka05** [onehilltech/backbone#1]
@hilljh82 This is more of a request than an issue. Hope you don't mind.

Upon starting to use DBFlow I took a look at the PR you were involved in: https://github.com/Raizlabs/DBFlow/pull/1091

I was hoping that it would be merged in soon, but since it doesnt look like that will happen anytime soon, I figured I would check out your project. 
So i landed here and looked through some of the source but im having trouble seeing how much of this fits together. It would be awesome if you had some documentation or even just a few examples

In particular im interested in the usage of Loaders and Cursors with DBFlow but im also interested in the backbone-data module you have here!

At a quick glance it appears to be an attempt to abstract web service calls / caching / and DBFlow handling out into a module.

**Comment by hilljh82** [comment]
@ka05 Sure, I can add some documentation for the Loaders and Cursors, which have not be merged into DBFlow.

Also, backbone is just a personal project that contains main components/classes that I reuse on many of my projects. Instead of making one single project, I have created several smaller projects to improve modularity.

In regards to the "web service calls / caching / and DBFlow handling out into a module" comment, the `backbone-data` project is my current attempt to implement [ember-data](https://github.com/emberjs/data) in Java. 

I personally grew tired of creating a backend api, and then having to implement the same code on Android for retrieving the data and storing the data in a SQLite database. So, `backbone-data` is designed to make it where you can define your data models in `DBFlow` (i.e., resources), and then make corresponding requests to an api server for those resources. The Cursors and Loaders that you speak about above make it where the views are automatically updated once the remote api call is complete. Right now, `backbone-data` use REST api format as explained in`ember-data` instead of JSONAPI. 

I will make sure to add documentation for `backbone-data`.

**Comment by hilljh82** [comment]
@ka05 I have added documentation in the form of a README

  https://github.com/onehilltech/backbone/tree/master/backbone-dbflow

**Comment by ka05** [comment]
@hilljh82 Thank you very much! And for being so quick to reply and take the time to write up the readme. I will read it over in more detail tomorrow. Thanks again.
